exporters/jaeger/CMakeLists.txt

```diff
@@ -26,6 +34,19 @@ if(MSVC)
   endif()
 endif()
 
+install(
```

Comment:
I am wondering whether we need install the header files for SDK exporters. I think we don't want any user to include `opentelemetry-cpp` via `find_package()` to include them? @open-telemetry/cpp-approvers

Reply:
How should we import SDK exporters then? I have a project with prebuilt opentelemetry-cpp and use these codes below to use opentelemetry-cpp.

```cmake
find_package(opentelemetry-cpp)

add_executable(example_otlp main.cc)
target_link_libraries(
  example_otlp
  opentelemetry-cpp::trace
  opentelemetry-cpp::otlp_exporter
  opentelemetry-cpp::ostream_span_exporter
  opentelemetry-cpp::api)
```

Reply:
The exported library target could be used for linking so it makes sense to install and export, but there is no need to install the header file as we don't expect the user to call the exporter functions directly?

Reply:
@ThomsonTan  - Won't user need to include these exporters headers too ( as here: https://github.com/open-telemetry/opentelemetry-cpp/blob/main/examples/simple/main.cc#L10 ). So they need to be installed?

Reply:
I found all examples use the headers of exporters(e.g. https://github.com/open-telemetry/opentelemetry-cpp/blob/main/examples/jaeger/main.cc#L22).
Is there any example about using a exporter without calling the exporter functions directly?

Reply:
You are right, the derived `SpanExporter` class definition should still be part of the header file. I was thinking about the `Recordable` stuff which is also installed by should not be directly visible to the user. I am fine to install the headers, perhaps we could move the non-public headers for exporter out of the installation path.

Reply:
I like the idea of hiding the code details about how to initialize SDK. Maybe we can provider a API in SDK to use a configure file to `initTracer()` in the future, and then all compoments are no need to exported then?
I'm not sure if this should be done.It's seems not any part of specification and the similar job is already done by `opentelemetry-collector` .

Reply:
Looking at the otlp, elasticsearch,  and jaeger exporter code,  we actually don't need to install their recordable.h, and so it can be excluded. Users would only use exporter headers. Zipkin exporter header is incorrectly including its recordable header, which needs to be fixed before excluding.
If we plan to handle this in separate PR, we can raise a issue for that.